### ISSUE TITLE
formatter: corrupts bare pseudo-class selector (`:link &`) into invalid CSS when it follows a template interpolation

### ISSUE THREAD

**Issue by techieyann** [oxc-project/oxc#24814]
### Input

```tsx
import {css, styled} from 'styled-components'

const Styled = styled.div<{status?: string}>`
  color: black;

  ${({status}) =>
    status === 'x' &&
    css`
      color: var(--palette-black);
    `}

  /* Add underline when inside a link */
  :link & {
    text-decoration: underline;
  }
`
```

### Config

```jsonc
{ "singleQuote": true, "semi": false }
```

Reproduces with default config too (semicolons/quote style aside).

### Oxfmt output

Oxfmt version: `0.58.0` (installed via `vite-plus@0.2.5`)

```tsx
import {css, styled} from 'styled-components'

const Styled = styled.div<{status?: string}>`
  color: black;

  ${({status}) =>
    status === 'x' &&
    css`
      color: var(--palette-black);
    `}/* Add underline when inside a link */
  : link & {
    text-decoration: underline;
  }
`
```

Two problems in one pass:
1. The blank line and the leading `/* comment */` get glued onto the interpolation's closing `` `} ``, losing the separation.
2. `:link &` — a bare pseudo-class selector nested with `&` — gets a space inserted after the colon, becoming `: link &`. That's invalid CSS; the browser drops the whole rule silently (no error, the `text-decoration: underline` behavior it was providing just disappears).

This isn't idempotent either: reformatting the already-corrupted output again leaves it unchanged (stable in the broken state), but formatting from a *correct* `:link &` input reliably reintroduces the bug — i.e. plain `vp fmt` (no flags) is a lossy round-trip for this pattern.

### Reduced trigger

The bug requires only "any `${...}` interpolation" directly before the comment + bare-pseudo-selector rule — it doesn't need the nested `css` tag or the conditional. This also reproduces:

```tsx
const Styled = styled.div<{status?: string}>`
  color: black;

  ${({status}) => status === 'x' && 'color: red;'}

  /* Add underline when inside a link */
  :link & {
    text-decoration: underline;
  }
`
```

Removing the interpolation entirely (leaving only the comment + `:link & { }` rule as the first thing in the template) formats correctly, which is what pointed us at the interpolation-adjacency as the trigger.

### Prettier output

Prettier version: `3.9.6`

```tsx
import { css, styled } from "styled-components";

const Styled = styled.div<{ status?: string }>`
  color: black;

  ${({ status }) =>
    status === "x" &&
    css`
      color: var(--palette-black);
    `}

  /* Add underline when inside a link */
  :link & {
    text-decoration: underline;
  }
`;
```

Prettier preserves the blank line, the comment placement, and the selector untouched.

### Additional notes

Found while upgrading a project from vite-plus 0.1.24 to 0.2.5 (oxfmt 0.52.0 → 0.58.0). The pattern is a normal styled-components idiom: a static pseudo-class rule (`&:hover`, `:link &`, etc.) placed near a dynamic `${...}` block for conditional styles. Worked around it in our codebase by reordering the static rule ahead of the interpolation, which avoids the trigger, but a plain `vp fmt`/`oxfmt --write` run anywhere else in a large codebase could silently drop real CSS rules the same way, with no warning or error — it just prints valid-looking (but semantically broken) code.

**Comment by leaysgur** [comment]
Thanks for the report. This is already fixed and released as Oxfmt v0.60.0.
